File: packages/sdk/src/videos.ts
/**
 * 视频模块 — 生成 / 任务查询 / 编排。
 *
 * 端点(3 个):
 * - POST /v1/videos/generations
 * - GET  /v1/videos/tasks/:id
 * - POST /v1/videos/compose
 */

import type { BaseClient } from './base.js'
import type {
  V1VideoGenerationsRequest,
  V1VideoGenerationsResponse,
  V1VideoTaskResponse,
  V1VideoComposeRequest,
  V1VideoComposeResponse,
} from '@ihui/types'

export interface VideosModule {
  /** POST /v1/videos/generations(视频生成,异步任务)。 */
  generations(req: V1VideoGenerationsRequest): Promise<V1VideoGenerationsResponse>
  /** GET /v1/videos/tasks/:id(查询视频任务状态)。 */
  getTask(taskId: string): Promise<V1VideoTaskResponse>
  /** POST /v1/videos/compose(视频编排)。 */
  compose(req: V1VideoComposeRequest): Promise<V1VideoComposeResponse>
}

export function createVideosModule(client: BaseClient): VideosModule {
  return {
    generations: (req) =>
      client.request<V1VideoGenerationsResponse>('POST', '/videos/generations', req),
    getTask: (taskId) =>
      client.request<V1VideoTaskResponse>('GET', `/videos/tasks/${encodeURIComponent(taskId)}`),
    compose: (req) => client.request<V1VideoComposeResponse>('POST', '/videos/compose', req),
  }
}

File: packages/sdk/python/ihui_ai/modules/videos.py
"""视频模块 — 生成 / 任务查询 / 编排。

端点(3 个):
- POST /v1/videos/generations
- GET  /v1/videos/tasks/:id
- POST /v1/videos/compose
"""

from __future__ import annotations

from urllib.parse import quote

from ..async_base import AsyncBaseClient
from ..base import BaseClient
from ..types import (
    V1VideoComposeRequest,
    V1VideoComposeResponse,
    V1VideoGenerationsRequest,
    V1VideoGenerationsResponse,
    V1VideoTaskResponse,
)


class VideosApi:
    """视频模块(同步)— 生成 / 任务查询 / 编排。"""

    def __init__(self, client: BaseClient) -> None:
        self._client = client

    def generations(self, req: V1VideoGenerationsRequest) -> V1VideoGenerationsResponse:
        """POST /v1/videos/generations(视频生成,异步任务)。"""
        return self._client.request("POST", "/videos/generations", req)

    def get_task(self, task_id: str) -> V1VideoTaskResponse:
        """GET /v1/videos/tasks/:id(查询视频任务状态)。"""
        return self._client.request("GET", f"/videos/tasks/{quote(task_id, safe='')}")

    def compose(self, req: V1VideoComposeRequest) -> V1VideoComposeResponse:
        """POST /v1/videos/compose(视频编排)。"""
        return self._client.request("POST", "/videos/compose", req)


class AsyncVideosApi:
    """视频模块(asyncio)— 生成 / 任务查询 / 编排。"""

    def __init__(self, client: AsyncBaseClient) -> None:
        self._client = client

    async def generations(self, req: V1VideoGenerationsRequest) -> V1VideoGenerationsResponse:
        """POST /v1/videos/generations(视频生成,异步任务)。"""
        return await self._client.request("POST", "/videos/generations", req)

    async def get_task(self, task_id: str) -> V1VideoTaskResponse:
        """GET /v1/videos/tasks/:id(查询视频任务状态)。"""
        return await self._client.request("GET", f"/videos/tasks/{quote(task_id, safe='')}")

    async def compose(self, req: V1VideoComposeRequest) -> V1VideoComposeResponse:
        """POST /v1/videos/compose(视频编排)。"""
        return await self._client.request("POST", "/videos/compose", req)


__all__ = ["VideosApi", "AsyncVideosApi"]
